app/com/lucidchart/open/cashy/controllers/UploadController.scala

```diff
@@ -90,40 +90,31 @@ class UploadController extends AppController {
               }
             }
 
-            // Check if gzip compression is appropriate
-            val compressedBytes = GzipHelper.compress(bytes)
-            val gzipSavings = (bytes.length - compressedBytes.length) / bytes.length.toDouble
-            val gzipUploaded = if (gzipSavings >= minGzipSavings) {
-              if(!S3Client.uploadToS3(bucket, assetName, compressedBytes, contentType, gzipped=true)) {
-                throw new UploadFailedException("Uploading gzip version for file failed: " + (bucket, assetName))
-              }
-              true
-            } else {
-              false
-            }
-
-            if(S3Client.uploadToS3(bucket, assetName, bytes, contentType)) {
+            val assets = mutable.Map[String,Asset]()
 
-              val assetId = AssetModel.createAsset(bucket, assetName, user.id)
-              val asset = AssetModel.findById(assetId).get
+            // If it is javascript and not already minified, try to minify it
+            if (checkExtension(extensions.js, assetName) && !checkMinified(assetName)) {
+              val extension = getExtension(assetName)
+              val minAssetName = assetName.substring(0, assetName.toLowerCase.lastIndexOf("." + extension.toLowerCase)) + ".min." + extension
 
-              AuditModel.createUploadAudit(user.id, bucket, assetName, asset.link, gzipUploaded)
+              // Make sure a min version with this name does not already exist
+              if(!S3Client.existsInS3(bucket, minAssetName)) {
+                val (minBytes, compressErrors) = JsCompress.compress(bytes)
```

Comment:
we should probably warn the user if minification isn't done cause there's already an asset with the minified name.